src/beamlime/handlers/config_handler.py

```diff
@@ -57,6 +77,17 @@ def handle(self, messages: list[Message[bytes]]) -> list[Message[None]]:
                     'Updating config: %s = %s at %s', key, value, message.timestamp
                 )
                 self._store[key] = value
-            except Exception as e:  # noqa: PERF203
-                self._logger.error('Error processing config message: %s', e)
+                updated[key] = value
+            except Exception:  # noqa: PERF203
+                self._logger.exception('Error processing config message:')
+        # Delay action calls until all messages are processed to reduce triggering
+        # multiple calls for the same key in case of multiple messages with same key.
+        for key, value in updated.items():
+            for action in self._actions.get(key, []):
+                try:
+                    action(value)
+                except KeyError:  # noqa: PERF203
+                    self._logger.exception(
+                        'Error processing config action for %s:', key
+                    )
```

Comment:
Not so important, but do we want to call actions only when it changes...? We could quickly check if the values are actually changed and then add it to `updated` dictionary.

Reply:
It's fine if it's expected to be handled even if it has same value.